digiroad2-viite/src/main/scala/fi/liikennevirasto/viite/ProjectValidator.scala

```diff
@@ -569,15 +569,45 @@ class ProjectValidator {
     )
   }
 
+//  def checkForInvalidUnchangedLinks(project: Project, projectLinks: Seq[ProjectLink]): Seq[ValidationErrorDetails] = {
+//    val invalidUnchangedLinks: Seq[ProjectLink] = projectLinks.groupBy(s => (s.roadNumber, s.roadPartNumber)).flatMap { g =>
+//      val (unchanged, others) = g._2.partition(_.status == UnChanged)
+//      //foreach number and part and foreach UnChanged found in that group, we will check if there is some link in some other different action, that is connected by geometry and addressM values to the UnChanged link starting point
+//      unchanged.groupBy(_.track) // Check per track
+//        .flatMap(upl => upl._2
+//          .filter(u => others.filterNot(opl => opl.status == LinkStatus.NotHandled || opl.track != upl._1)
+//            .exists(o => u.startAddrMValue >= o.startAddrMValue)))
+//    }.toSeq
+//
+//    if (invalidUnchangedLinks.nonEmpty) {
+//      Seq(error(project.id, ValidationErrorList.ErrorInValidationOfUnchangedLinks)(invalidUnchangedLinks).get)
+//    } else {
+//      Seq()
+//    }
+//  }
+
   def checkForInvalidUnchangedLinks(project: Project, projectLinks: Seq[ProjectLink]): Seq[ValidationErrorDetails] = {
```

Comment:
Voinks mää rutista tonne funktiokommentaarin, jotain tähän tyyliin:
/** The Unchanged status is allowed only for the links whose <i>address information does not change</i> within the project.
    Any operation, other  than Unchanged ("Ennallaan"), applied to the link, changes the addresses of the links following the operation. Thus, the links following the first other operation must not be allowed to have "Unchanged" status, but is returnad with a validation error.
*/